From: Oleksij Rempel <o.rempel@pengutronix.de>
Date: Thu, 22 Sep 2022 10:55:03 +0200
Subject: [PATCH] net: dsa: do not wait for aneg is actually done on the port
 start

Aneg configuration should be done by the eth code, not dsa.

Signed-off-by: Oleksij Rempel <o.rempel@pengutronix.de>
---
 drivers/net/dsa.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/drivers/net/dsa.c b/drivers/net/dsa.c
index 7a64a7941223..040ba897e256 100644
--- a/drivers/net/dsa.c
+++ b/drivers/net/dsa.c
@@ -111,10 +111,6 @@ static int dsa_port_start(struct eth_device *edev)
 
 	dsa_port_set_ethaddr(edev);
 
-	ret = phy_wait_aneg_done(dp->edev.phydev);
-	if (ret)
-		return ret;
-
 	if (ops->port_enable) {
 		ret = ops->port_enable(dp, dp->index, dp->edev.phydev);
 		if (ret)
